internal/api_server/middleware/tls.go

```diff
@@ -20,10 +20,12 @@ const TLSCommonNameContextKey contextKey = "tls-cn"
 
 func NewHTTPServer(router http.Handler, log logrus.FieldLogger, address string) *http.Server {
 	return &http.Server{
-		Addr:         address,
-		Handler:      router,
-		ReadTimeout:  5 * time.Second,
-		WriteTimeout: 10 * time.Second,
+		Addr:              address,
+		Handler:           router,
+		ReadTimeout:       5 * time.Second,
+		ReadHeaderTimeout: 5 * time.Second,
+		WriteTimeout:      10 * time.Second,
+		MaxHeaderBytes:    4096,
```

Comment:
Comparing notes with [K8s's API server config(https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/secure_serving.go#L165-L173) and [kubelet's](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/server.go#L176-L183) they define IdleTimeout and set much higher values for MaxHeaderBytes (though IDK of a situation we'd ever have such large headers?JWTs? ) and timeouts.

Reply:
4 hours for ReadTimeout?  Odd.
I assume we'll change these values a few times along the way.  We just need something sane to start with.

Reply:
Should we make all this configurable parameters? (with sensible defaults, of course)

I am just worried that under unexpected circumstances this could not work out on the field, and then customers have no escape hatch.

Reply:
The timeouts are probably very low, and problematic for console over HTTP or gRPC over HTTP as we want to do, I'd at least bump those at least an order of magnitude

Reply:
Set all timeouts to 5 minutes by default, and headers to 32KB.  All configurable.